packages/core/src/processors/runner.ts

```diff
@@ -0,0 +1,291 @@
+import type { ObjectStreamPart, StreamObjectResult, TextStreamPart } from 'ai';
+import type { MastraMessageV2, MessageList } from '../agent/message-list';
+import { TripWire } from '../agent/trip-wire';
+import type { StreamTextResult } from '../llm';
+import type { IMastraLogger } from '../logger';
+import type { Processor } from './index';
+
+/**
+ * Implementation of processor state management
+ */
+class ProcessorState {
+  private accumulatedText = '';
+  private customState = new Map<string, any>();
+  public allChunks: (TextStreamPart<any> | ObjectStreamPart<any>)[] = [];
+
+  constructor(private readonly processorName: string) {}
+
+  // Internal methods for the runner
+  addChunk(chunk: TextStreamPart<any> | ObjectStreamPart<any>): void {
+    // Extract text from text-delta chunks for accumulated text
+    if (chunk.type === 'text-delta') {
+      this.accumulatedText += chunk.textDelta;
+    }
+    this.allChunks.push(chunk);
+  }
+
+  // Public interface for processors
+  getState(): Record<string, any> {
+    return {
+      ...Object.fromEntries(this.customState),
+    };
+  }
+
+  // Helper methods for backward compatibility
+  get<T>(key: string): T | undefined {
+    return this.customState.get(key);
+  }
+
+  set<T>(key: string, value: T): void {
+    this.customState.set(key, value);
+  }
```

Comment:
this should just be an empty object they can mutate, no reason to make it more complicated